sale_order_product_recommendation/wizards/sale_order_recommendation.py

```diff
@@ -311,21 +309,24 @@ def _compute_price_unit(self):
             else:
                 line.price_unit = line._get_last_sale_price_product()
 
-    def _prepare_update_so_line(self, line_form):
-        """So we can extend SO update"""
+    def _prepare_update_so_line_vals(self):
+        vals = {"product_uom_qty": self.units_included}
         if not self.product_uom_readonly:
-            line_form.product_uom = self.sale_uom_id
-        line_form.product_uom_qty = self.units_included
+            vals.update({"product_uom": self.sale_uom_id.id})
```

Comment:
Cleaner syntax when it's only one element:

```suggestion
            vals["product_uom"] = self.sale_uom_id.id
```